pkg/services/ngalert/api/api_ruler.go

```diff
@@ -449,6 +449,7 @@ func (srv RulerSrv) updateAlertRulesInGroup(c *models.ReqContext, groupKey ngmod
 }
 
 func toGettableRuleGroupConfig(groupName string, rules []*ngmodels.AlertRule, namespaceID int64, provenanceRecords map[string]ngmodels.Provenance) apimodels.GettableRuleGroupConfig {
+	ngmodels.RulesGroup(rules).SortByGroupIndex()
```

Comment:
Are we creating a new type just to make the sorting function take zero arguments? Do you intend to use `RulesGroup` in more cases in future changes? I'd prefer writing a function with no receiver unless we have other uses for the type, because to a naive reader this looks like two chained function calls instead of a type conversion.

Reply:
Idea is to distinguish between an arbitrary slice of AlertRule and a slice that contains rules that belong to the same group.

Reply:
I updated APIs to use this alias. Also, it will be used in the scheduler